src/dmd/target.d

```diff
@@ -849,3 +719,164 @@ private Expression decodeReal(const ref Loc loc, Type type, ubyte* buffer)
     }
     return new RealExp(loc, value, type);
 }
+
+/**
+ * Params:
+ *   tf = function type to check
+ *   fd = function to check (optional)
+ * Returns:
+ *   true if return value from function is on the stack
+ */
+private bool isReturnOnStackImpl(TypeFunction tf, FuncDeclaration fd)
```

Comment:
It doesn't look like you need `FunctionDeclaration`, you just need `fd.needThis()`.  You could potentially make it a `bool needsThis` to remove the dependency.